packages/lib/server-only/user/verify-email.ts

```diff
@@ -0,0 +1,54 @@
+import { prisma } from '@documenso/prisma';
+
+import { generateConfirmationToken } from './generate-confirmation-token';
+
+export type VerifyEmailProps = {
+  token: string;
+};
+
+export const verifyEmail = async ({ token }: VerifyEmailProps) => {
+  const dbToken = await prisma.verificationToken.findFirst({
+    include: {
+      user: true,
+    },
+    where: {
+      token,
+    },
+  });
+
+  if (!dbToken) {
+    throw new Error('Invalid token provided. Please try again.');
+  }
+
+  // check if the token is valid or expired
+  const valid = dbToken.expires > new Date();
+
+  if (!valid) {
+    // if the token is expired, generate a new token and send the email
+    // and return false
+    await generateConfirmationToken({ email: dbToken.user.email });
+    return valid;
+  }
+
+  const [updatedUsers, deletedToken] = await prisma.$transaction([
+    prisma.user.update({
+      where: {
+        id: dbToken.userId,
+      },
+      data: {
+        emailVerified: new Date(),
+      },
+    }),
+    prisma.verificationToken.deleteMany({
+      where: {
+        userId: dbToken.userId,
+      },
+    }),
+  ]);
+
+  if (!updatedUsers || !deletedToken) {
```

Comment:
If this is returning an array of users it will always be truthy, consider how we might better handle this.

Reply:
For some reason, I created `updatedUsers` even though it's updating only one user. My bad. I'll also think of a better way to handle this.